ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/bnd/TargetRepository.java

```diff
@@ -303,6 +309,35 @@ public InputStream getContent() {
 			throw new RuntimeException(new FileNotFoundException());
 		}
 
+		@Override
+		public <T> T getAdapter(Class<T> adapter) {
+			if (adapter == BundleDescription.class) {
+				return adapter.cast(bundle);
+			}
+			if (adapter == File.class) {
+				String location = bundle.getLocation();
+				if (location != null) {
+					File file = new File(location);
+					if (file.isFile()) {
+						return adapter.cast(file);
+					}
+				}
+			}
+			if (adapter == Path.class) {
+				String location = bundle.getLocation();
+				if (location != null) {
+					Path path = Path.of(location);
+					if (Files.isRegularFile(path)) {
+						return adapter.cast(path);
+					}
+				}
+			}
+			if (adapter == Repository.class) {
+				return adapter.cast(repository);
+			}
```

Comment:
Are the adapters to File, Path and Repo are currently in use?

Reply:
Not yet but they are handy for further improvements.

Reply:
Then please let's add this as soon as it is used.
In case it isn't used an unlucky person will have a hard time in the future to find out if this is used or not.